crates/sui-core/src/authority.rs

```diff
@@ -2161,13 +2157,67 @@ impl AuthorityState {
         certificate.verify(&self.committee.load())
     }
 
+    /// Verifies transaction signatures and other data
+    /// Important: This function can potentially be called in parallel and you can not rely on order of transactions to perform verification
+    /// If this function return an error, transaction is skipped and is not passed to handle_consensus_transaction
+    /// This function returns unit error and is responsible for emitting log messages for internal errors
+    pub(crate) fn verify_consensus_transaction(
+        &self,
+        transaction: SequencedConsensusTransaction,
+    ) -> Result<VerifiedSequencedConsensusTransaction, ()> {
+        let _timer = self
+            .metrics
+            .verify_narwhal_transaction_duration_mcs
+            .utilization_timer();
+        let committee = self.committee.load();
+        match &transaction.transaction.kind {
+            ConsensusTransactionKind::UserTransaction(certificate) => {
+                if self
+                    .database
+                    .consensus_message_processed(certificate.digest())
+                    .expect("Storage error")
```

Comment:
Just to be sure: panicking here ensures that the consensus transaction will not be skipped (ie. validator stops/crashes and the execution indices do not update), right?

Reply:
I think in general when we have storage error there is not much we can do. Yes, this will not skip transaction, but essentially will shutdown the node